syncd/scripts/syncd_init_common.sh

```diff
@@ -124,74 +124,188 @@ config_syncd_cisco_8000()
+                      echo "Override the config $(grep $sedline $to_file) with $line in $to_file"
+                      prop=${line:0:`expr index "$line" :`}
+                      sed -i "/$prop/d" $to_file
+                      echo "        $line" >> $to_file
+                      merged_cnt+=1
                    else
-                       sedline=${line%=*}
-                       if grep -q $sedline $file ;then
-                          echo "Keep the config $(grep $sedline $file) in $file"
-                       else
-                          echo $line >> $file
-                       fi
+                      grepline=$(grep $sedline $to_file)
+                      grepline="${grepline#"${grepline%%[![:space:]]*}"}"
+                      if [ "${grepline::1}" == '#' ];then
+                         echo "        $line" >> $to_file
+                         merged_cnt+=1
+                      else
+                         echo "Keep the config $(grep $sedline $to_file) in $to_file"
+                      fi
                    fi
-               fi
-             done < $COMMON_CONFIG_BCM
-             echo "# End of chip common properties" >> $file
-         done
-         echo "Merging $PLT_CONFIG_BCM with $COMMON_CONFIG_BCM, merge files stored in $CONFIG_BCM"
-      fi
-
-      #sync the file system
-      sync
-
-      # copy the final config.bcm and sai.profile to the shared folder for 'show tech'
-      cp -f /tmp/sai.profile /var/run/sswsyncd/
-      cp -f /tmp/*.bcm /var/run/sswsyncd/
-
-      if [ -f "/tmp/sai.profile" ]; then
-          CMD_ARGS+=" -p /tmp/sai.profile"
-      elif [ -f "/etc/sai.d/sai.profile" ]; then
-          CMD_ARGS+=" -p /etc/sai.d/sai.profile"
-      else
-          CMD_ARGS+=" -p $HWSKU_DIR/sai.profile"
-      fi
+                else
+                   echo "        $line" >> $to_file
+                   merged_cnt+=1
+                fi
+            fi
+        fi
+    done < $from_file
 
-    else
+    if [ $merged_cnt -gt 0 ]; then
+         sed -i "/# Start of/a \    global:" $to_file
+         sed -i "/# Start of/a \  0:" $to_file
+         sed -i "/# Start of/a \bcm_device:" $to_file
+         sed -i "/# Start of/a \---" $to_file
+    fi
+    echo "# End of $message" >> $to_file
+    if [ $merged_cnt -gt 0 ]; then
+       sed -i "/# End of/i \..." $to_file
+    fi
+    echo "Merged $from_file to $to_file"
+}
+
+config_syncd_bcm()
+{
+    PLATFORM_COMMON_DIR=/usr/share/sonic/device/x86_64-broadcom_common
+    PLT_CONFIG_BCM=""
+    PLT_CONFIG_YML=""
+    PLT_SAI_PROFILE=$(find $HWSKU_DIR -name 'sai.profile')
+    readline=$(grep SAI_INIT_CONFIG_FILE $PLT_SAI_PROFILE)
+    if [ ${readline: -3} == "bcm" ]; then
+       PLT_CONFIG_BCM=${readline#*=}
+    elif [ ${readline: -3} == "yml" ]; then
+       PLT_CONFIG_YML=${readline#*=}
+    fi
+    
+    if [ ! -z "$PLT_CONFIG_BCM" ] && [ -f $PLATFORM_DIR/common_config_support ] ; then
+       cp -f $HWSKU_DIR/*.config.bcm /tmp
```

Comment:
If there is no *.config.bcm doesn't exist, this command will fail. Need to check  /etc/said.d/profile and find out where is the config.bcm file then copy it to /tmp

Reply:
So far the script does search the sai.profile to find platform config. 
The following copied escaped from the fix. 

Can you try the following change for line#250 to see if works ? I will fix it in the PR. 

cp -f $HWSKU_DIR/*.config.bcm /tmp => cp -f $PLT_CONFIG_BCM /tmp

Geans